pgcontents/tests/test_synchronization.py

```diff
@@ -177,3 +184,99 @@ def check_reencryption(old, new):
         # crypto manager.
         unencrypt_all_users(engine, crypto2_factory, logger)
         check_reencryption(manager2, no_crypto_manager)
+
+
+class TestGenerateNotebooks(TestCase):
+
+    def setUp(self):
+        remigrate_test_schema()
+
+    def tearDown(self):
+        clear_test_db()
+
+    def test_generate_notebooks(self):
+        """
+        Creates files and checkpoints for two users and checks their
+        accessibility through `generate_files` and `generate_checkpoints`.
+        """
+        db_url = TEST_DB_URL
+        engine = create_engine(db_url)
+        user_ids = ['test_generate_notebooks0', 'test_generate_notebooks1']
+        encryption_pw = u'foobar'
+
+        crypto_factory = single_password_crypto_factory(encryption_pw)
+        managers = {}
```

Comment:
Often the pattern of "make an empty container, then fill it by mapping an expression over an iterator" is better expressed as a list/set/dictionary comprehension.  In this case, that would look like:

```python
managers = {
    user_id: PostgresContentsManager(
        user_id=user_id,
        db_url=db_url,
        crypto=crypto_factory(user_id),
        create_user_on_startup=True,
    )
    for user in users
}
```
In this case, that's a bit of a mouthful because the `PostgresContentsManager` constructor is pretty verbose.  One way to improve that might be to factor out the particular invocation pattern here.

```python
def encrypted_pgmanager(db_url, user_id, crypto_factory):
    return PostgresContentsManager(user_id=user_id, crypto=crypto_factory(user_id), ...)
```